pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -0,0 +1,152 @@
+	Expect(err).ToNot(HaveOccurred(), "Failed to create raw config ")
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(rawConfig)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create rest config ")
+
+	client, err := client.New(restConfig, client.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred(), "Failed to create a cluster client")
+
+	return client, nil
+}
+
+func getClusterBootstrap(ctx context.Context, k8sClient client.Client, namespace, clusterName string) *runtanzuv1alpha3.ClusterBootstrap {
+	clusterBootstrap := &runtanzuv1alpha3.ClusterBootstrap{}
+	objKey := client.ObjectKey{Namespace: namespace, Name: clusterName}
+
+	Eventually(func() bool {
+		err := k8sClient.Get(ctx, objKey, clusterBootstrap)
+		return err == nil
+	}, waitTimeout, pollingInterval).Should(BeTrue())
+
+	Expect(clusterBootstrap).ShouldNot(BeNil())
+	return clusterBootstrap
+}
+
+func getPackageInstall(ctx context.Context, k8sClient client.Client, namespace, pkgiName string) *kapppkgiv1alpha1.PackageInstall {
+	pkgInstall := &kapppkgiv1alpha1.PackageInstall{}
+	objKey := client.ObjectKey{Namespace: namespace, Name: pkgiName}
+
+	Eventually(func() bool {
+		err := k8sClient.Get(ctx, objKey, pkgInstall)
+		return err == nil
+	}, waitTimeout, pollingInterval).Should(BeTrue())
+
+	Expect(pkgInstall).ShouldNot(BeNil())
+
+	return pkgInstall
+}
+
+func getPackageDetailsFromCBS(CBSRefName string) (string, string, string, error) {
+	pkgShortName := strings.Split(CBSRefName, ".")[0]
+
+	pkgName := strings.Join(strings.Split(CBSRefName, ".")[0:4], ".")
+
+	pkgVersion := strings.Join(strings.Split(CBSRefName, ".")[4:], ".")
+
+	return pkgShortName, pkgName, pkgVersion, nil
+}
+
+func checkClusterCBS(ctx context.Context, mccl, wccl client.Client, mcClusterName, wcClusterName, infrastructureName string) error {
+	var err error
+
+	// Get ClusterBootstrap and return error if not found
+	clusterBootstrap := getClusterBootstrap(ctx, mccl, constants.TkgNamespace, mcClusterName)
+
+	// get package details from clusterBootstrap
+	cniPkgShortName, cniPkgName, cniPkgVersion, err := getPackageDetailsFromCBS(clusterBootstrap.Spec.CNI.RefName)
+	Expect(err).NotTo(HaveOccurred())
+	kappPkgShortName, kappPkgName, kappPkgVersion, err := getPackageDetailsFromCBS(clusterBootstrap.Spec.Kapp.RefName)
+	Expect(err).NotTo(HaveOccurred())
+
+	// verify clusterBootstrap details matches package install on workload cluster
+	verifyPackageInstall(ctx, wccl, wcClusterName, cniPkgShortName, cniPkgName, cniPkgVersion)
+	verifyPackageInstall(ctx, wccl, wcClusterName, kappPkgShortName, kappPkgName, kappPkgVersion)
+
+	// loop over additional packages list in clusterBootstrap spec to check package information
+	if clusterBootstrap.Spec.AdditionalPackages != nil {
+		// validate additional packages
+		for _, additionalPkg := range clusterBootstrap.Spec.AdditionalPackages {
+			if additionalPkg.RefName == "metrics-server.tanzu.vmware.com.0.5.1+vmware.1-tkg.2-zshippable" {
```

Comment:
Why do we need this refname == "metrics-server.tanzu.vmware.com.0.5.1+vmware.1-tkg.2-zshippable" check ? Adding this makes it not generic. When we bump package we have to come here and change.

cant we just do this:

```
for _, additionalPkg := range clusterBootstrap.Spec.AdditionalPackages {
    pkgShortName, pkgName, pkgVersion, err := getPackageDetailsFromCBS(additionalPkg.RefName)
    Expect(err).NotTo(HaveOccurred())
    verifyPackageInstall(ctx, wccl, wcClusterName, pkgShortName, pkgName, pkgVersion)
}
```

Reply:
that's a good suggestion! I've updated this part in the latest commit